### PR TITLE
Check docs v2

### PR DESCRIPTION
**Pull request by AydinHassan**
Not done the listener check docs yet, will add in a further PR - not really sure what to build their as an example - any thoughts @mikeymike ?

Currently the only listener check we have is the database check.

### CODE REVIEW

File: templates/docs/reference/bundled-checks.phtml [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+</code></pre>
+
+<p>The output would be something along the lines of:</p>
+<a href="/img/func-requirements.png"><img src="/img/func-requirements.png" class="doc-terminal-screen"></a>
+
+<?= $this->renderContentHeader('composer-check', 'Composer Check') ?>
+<p>Here is an example of how to force the student to require the <code>nikic/fast-route</code> package via Composer. This
+is useful if you want to focus on a specific problem or promote popular/battle tested packages.</p>
+
+<pre><code class="lang-php">class MyExercise extends AbstractExercise implements ExerciseInterface, ComposerExerciseCheck
```
Comment (mikeymike) [inline templates/docs/reference/bundled-checks.phtml]: @AydinHassan what do you think about putting the implements on newlines, they don't fit 100% in the sites container and it's not really explicit on all browsers that it's a scroll until you actually scroll 

http://ss.jhf.tw/Screen%20Shot%202016-05-16%20at%2021.38.32-H3FiMnbcIK.png

**Comment by mikeymike** [comment]
@AydinHassan RE the listener check... I get what you mean 😞  Maybe just use the database check as the example ?

File: templates/docs/reference/creating-simple-checks.phtml [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+<p>Creating an check begins with creating a file and a class for our check . We need to implement the interface
+    <code>PhpSchool\PhpWorkshop\Check\SimpleCheckInterface</code> which extends from
+    <code>PhpSchool\PhpWorkshop\Check\CheckInterface</code>. Let's breakdown these methods before we start coding:</p>
+
+<h3><code>getName()</code></h3>
+<p>This method should just return a <code>string</code> which represents the name of the check. This will be printed on
+    the terminal during the verification process. This will be <code>PSR2 Code Check</code> for our check.</p>
+
+<h3><code>getExerciseInterface()</code></h3>
+<p>This method should just return a <code>string</code> which is the FQCN of the interface that the exercise needs
```
Comment (mikeymike) [inline templates/docs/reference/creating-simple-checks.phtml]: should we assume everyone is familiar with abbreviations? (FQCN) in this case

File: app/config.php [hunk 3]
```diff
@@ -93,11 +93,13 @@
         $referenceGroup->addSection('available-services', 'Available Services', 'docs/reference/available-services.phtml');
         $referenceGroup->addSection('exercise-types', 'Exercise Types', 'docs/reference/exercise-types.phtml');
         $referenceGroup->addSection('exercise-solutions', 'Exercise Solutions', 'docs/reference/exercise-solutions.phtml');
-        $referenceGroup->addSection('self-checking-exercises', 'Self Checking Exercises', 'docs/reference/self-checking-exercises.phtml', false);
+        $referenceGroup->addSection('exercise-checks', 'Exercise Checks', 'docs/reference/exercise-checks.phtml');
+        $referenceGroup->addSection('bundled-checks', 'Bundled Checks', 'docs/reference/bundled-checks.phtml');
+        $referenceGroup->addSection('creating-simple-checks', 'Creating Simple Checks', 'docs/reference/creating-simple-checks.phtml');
+        $referenceGroup->addSection('creating-listener-checks', 'Creating Listener Checks', 'docs/reference/creating-listener-checks.phtml', false);
```
Comment (mikeymike) [inline app/config.php]: noticed even though it's false it still has the link in the pagination ... I'm assuming a bug ?

File: templates/docs/reference/self-checking-exercises.phtml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+     */
+    public function check($fileName)
+    {
+        //do some checking with $fileName
+
+        if ($someResult) {
+            return new Success('My Check');
+        }
+
+        return new Failure('My Check', 'Something didn't go well!');
```
Comment (mikeymike) [inline templates/docs/reference/self-checking-exercises.phtml]: escape the `'` or use `"` 👍

**Comment by mikeymike** [comment]
Hey man this is great stuff ❤️

File: app/config.php (hunk 3 above)
Comment (AydinHassan) [inline app/config.php]: yeah, it's a bug. will fix after.

File: templates/docs/reference/creating-simple-checks.phtml (hunk 2 above)
Comment (AydinHassan) [inline templates/docs/reference/creating-simple-checks.phtml]: nope, updated to include in full noq.

File: templates/docs/reference/bundled-checks.phtml (hunk 1 above)
Comment (AydinHassan) [inline templates/docs/reference/bundled-checks.phtml]: yep, done :)

File: templates/docs/reference/self-checking-exercises.phtml (hunk 4 above)
Comment (AydinHassan) [inline templates/docs/reference/self-checking-exercises.phtml]: done

**Pull request merged**